options.js

```diff
@@ -88,88 +94,123 @@ function updateShortcutInputText(inputId, keyCode) {
-  rewindKeyCode = isNaN(rewindKeyCode) ? tcDefaults.rewindKeyCode : rewindKeyCode;
-  advanceKeyCode = isNaN(advanceKeyCode) ? tcDefaults.advanceKeyCode : advanceKeyCode;
-  slowerKeyCode = isNaN(slowerKeyCode) ? tcDefaults.slowerKeyCode : slowerKeyCode;
-  fasterKeyCode = isNaN(fasterKeyCode) ? tcDefaults.fasterKeyCode : fasterKeyCode;
-  fastKeyCode   = isNaN(fastKeyCode) ? tcDefaults.fastKeyCode : fastKeyCode;
+  var blacklist = document.getElementById('blacklist').value;
+  var version = tcDefaults.version;
+
   displayKeyCode = isNaN(displayKeyCode) ? tcDefaults.displayKeyCode : displayKeyCode;
 
+  chrome.storage.sync.clear();
   chrome.storage.sync.set({
-    speedStep:      speedStep,
-    rewindTime:     rewindTime,
-    advanceTime:    advanceTime,
-    fastSpeed:      fastSpeed,
-    resetKeyCode:   resetKeyCode,
-    rewindKeyCode:  rewindKeyCode,
-    advanceKeyCode: advanceKeyCode,
-    slowerKeyCode:  slowerKeyCode,
-    fasterKeyCode:  fasterKeyCode,
-    fastKeyCode:    fastKeyCode,
     displayKeyCode: displayKeyCode,
-    rememberSpeed:  rememberSpeed,
-    startHidden:    startHidden,
-    blacklist:      blacklist.replace(/^\s+|\s+$/gm,'')
-  }, function() {
+    rememberSpeed: rememberSpeed,
+    startHidden: startHidden,
+    keyBindings: keyBindings,
+    version: version,
+    blacklist: blacklist.replace(/^\s+|\s+$/gm, '')
+  }, function () {
     // Update status to let user know options were saved.
     var status = document.getElementById('status');
     status.textContent = 'Options saved';
-    setTimeout(function() {
+    setTimeout(function () {
       status.textContent = '';
     }, 1000);
   });
 }
 
 // Restores options from chrome.storage
 function restore_options() {
-  chrome.storage.sync.get(tcDefaults, function(storage) {
-    document.getElementById('speedStep').value = storage.speedStep.toFixed(2);
-    document.getElementById('rewindTime').value = storage.rewindTime;
-    document.getElementById('advanceTime').value = storage.advanceTime;
-    document.getElementById('fastSpeed').value = storage.fastSpeed;
-    updateShortcutInputText('resetKeyInput',  storage.resetKeyCode);
-    updateShortcutInputText('rewindKeyInput', storage.rewindKeyCode);
-    updateShortcutInputText('advanceKeyInput', storage.advanceKeyCode);
-    updateShortcutInputText('slowerKeyInput', storage.slowerKeyCode);
-    updateShortcutInputText('fasterKeyInput', storage.fasterKeyCode);
-    updateShortcutInputText('fastKeyInput', storage.fastKeyCode);
+  chrome.storage.sync.get(tcDefaults, function (storage) {
     updateShortcutInputText('displayKeyInput', storage.displayKeyCode);
     document.getElementById('rememberSpeed').checked = storage.rememberSpeed;
     document.getElementById('startHidden').checked = storage.startHidden;
     document.getElementById('blacklist').value = storage.blacklist;
+
+    for (let i in storage.keyBindings) {
+      var item = storage.keyBindings[i];
+      if (i < 6) {
```

Comment:
Instead of relying on indexes (this will break and cause subtle bugs), let's annotate the keybindings with "default" or "visible" flag.

Reply:
that way we should also check for default flag for buttons (which is not working now)
what do you mean with visible? visible in overlay? It might be good that way

Reply:
'visible' might not be the best name. I just mean that we shouldn't rely on position indexes and instead add an explicit flag for any shortcuts that have special treatment. We can use 'required', or 'default', or ...